airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/utils/connector_runner.py

```diff
@@ -208,6 +210,18 @@ async def call_read_with_state(
             enable_caching=enable_caching,
         )
 
+    async def do_setup(self, dagger_client: dagger.Client, container: dagger.Container, config: SecretDict, setup_script_path: Path):
```

Comment:
The limitation I'm foreseeing here is that all dependency used in the script (let's say an SQL client to connect to the db) has to be available in the connector image, which is not guaranteed. This is why I would suggest using a Dockerfile and `dagger_client.container().build("<path-to-dockerfile">)`.

This is not blocking to me as long as DB source's team setup/teardown can be executed in the connector container...